docs/rfcs/rfc-030-remove-runFirstColumn-from-aggregate-where.md

```diff
@@ -0,0 +1,70 @@
+# Remove `runFirstColumn` from Aggregate Where
+
+## Problem
+
+We are aiming to remove `apoc.runFirstColumn` use from our code base because it causes the following issues:
+* Hard to maintain and compose queries
+* Bad performance due to secondary processes being spawned for the queries
+* apoc methods cannot be properly planned by Neo4j database, leading to worse performance.
+
+## Current Solution
+
+Currently, for the following type definitions:
+
+```graphql
+    type User {
+        name: String!
+    }
+
+    type Post {
+        content: String!
+        likes: [User!]! @relationship(type: "LIKES", direction: IN)
+    }
+```
+
+The following query:
+
+```graphql
+{
+	posts(where: { likesAggregate: { count: 10 } }) {
+		content
+	}
+}
+```
+
+Would produce the following cypher:
+
+```cypher
+MATCH (this:Post)
+WHERE apoc.cypher.runFirstColumnSingle(
+    "MATCH (this)<-[aggr_edge:LIKES]-(aggr_node:User)RETURN count(aggr_node) = 10",
+    { this: this, aggr_count: 10 }
+)
+RETURN this { .content } AS this'
+```
+
+## Proposed Solution
+
+Use a `CALL` clause to perform the query that is currently performed within `runFirstColumn`.
+
+Therefore, the example query should produce the following cypher:
+
+```cypher
+MATCH(this:Post)
+CALL {
+    WITH this
+    MATCH(this)<-[:LIKES]-(u:User)
+    RETURN count(u) as agg_count
+}
+WITH *
+WHERE agg_count = 10
+RETURN this { .content } AS this
```

Comment:
Not blocking, 
I would suggest keeping the name of the cypher variable more dependent on the context used so that a future implementation that uses this RFC as documentation could be less prone to introduce a variable name collision.

```suggestion
MATCH(this:Post)
CALL {
    WITH this
    MATCH(this)<-[:LIKES]-(u:User)
    RETURN count(u) as thisLikesAggregateCount
}
WITH *
WHERE thisLikesAggregateCount = 10
RETURN this { .content } AS this
```